### PR TITLE
Microsoft Vendor Specific Attributes - unicodedecode error fix

### PR DESCRIPTION
**Pull request by kumar35959**
Fix for `microsoft vendor specific attributes`, where value is not utf-8 string.

Testing:

Reproduced and tested on local setup.

### CODE REVIEW

File: src/modules/rlm_python3/rlm_python3.c [hunk 1]
```diff
@@ -417,9 +417,21 @@ static int mod_populate_vptuple(PyObject *pPair, VALUE_PAIR *vp)
 	if (pStr == NULL) {
 		ERROR("%s:%d, vp->da->name: %s", __func__, __LINE__, vp->da->name);
 		if (PyErr_Occurred()) {
-			python_error_log();
+			if (PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)) {
+				python_error_log();
```
Comment (aren) [inline src/modules/rlm_python3/rlm_python3.c]: Maybe just a log line about how the string isn't UTF8 so it's being returned as a byte string instead? I don't think we should expose the exception.

Comment (kumar35959) [reply src/modules/rlm_python3/rlm_python3.c]: Done. Cleared the error, without logging the error

File: src/modules/rlm_python3/rlm_python3.c [hunk 2]
```diff
@@ -417,9 +417,21 @@ static int mod_populate_vptuple(PyObject *pPair, VALUE_PAIR *vp)
 	if (pStr == NULL) {
 		ERROR("%s:%d, vp->da->name: %s", __func__, __LINE__, vp->da->name);
 		if (PyErr_Occurred()) {
-			python_error_log();
+			if (PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)) {
+				PyErr_Clear();
```
Comment (aren) [inline src/modules/rlm_python3/rlm_python3.c]: Can you print a debugging message here (not an ERROR), like radlog("Conversion to Unicode failed, returning as %s as bytes", vp->da->name).

I didn't use radlog correctly, but you get the idea.

File: src/modules/rlm_python3/rlm_python3.c [hunk 3]
```diff
@@ -417,9 +417,21 @@ static int mod_populate_vptuple(PyObject *pPair, VALUE_PAIR *vp)
 	if (pStr == NULL) {
 		ERROR("%s:%d, vp->da->name: %s", __func__, __LINE__, vp->da->name);
```
Comment (aren) [inline src/modules/rlm_python3/rlm_python3.c]: move this down to line 431 so it's only shown if it wasn't a UnicodeDecodeError

File: src/modules/rlm_python3/rlm_python3.c (hunk 2 above)
Comment (kumar35959) [reply src/modules/rlm_python3/rlm_python3.c]: done

File: src/modules/rlm_python3/rlm_python3.c (hunk 3 above)
Comment (kumar35959) [reply src/modules/rlm_python3/rlm_python3.c]: done

**Pull request merged**